op-challenger/cmd/create_game.go

```diff
@@ -0,0 +1,137 @@
+)
+
+const envVarPrefix = "OP_CHALLENGER"
+
+var (
+	TraceTypeFlag = &cli.StringFlag{
+		Name:    "trace-type",
+		Usage:   "Trace types to support.",
+		EnvVars: opservice.PrefixEnvVar(envVarPrefix, "TRACE_TYPE"),
+		Value:   config.TraceTypeCannon.String(),
+	}
+	OutputRootFlag = &cli.StringFlag{
+		Name:    "output-root",
+		Usage:   "The output root for the fault dispute game.",
+		EnvVars: opservice.PrefixEnvVar(envVarPrefix, "OUTPUT_ROOT"),
+	}
+	L2BlockNumFlag = &cli.StringFlag{
+		Name:    "l2-block-num",
+		Usage:   "The l2 block number for the game.",
+		EnvVars: opservice.PrefixEnvVar(envVarPrefix, "L2_BLOCK_NUM"),
+	}
+)
+
+func CreateGame(ctx *cli.Context) error {
+	logger, err := setupLogging(ctx)
+	if err != nil {
+		return err
+	}
+	rpcUrl := ctx.String(flags.L1EthRpcFlag.Name)
+	if rpcUrl == "" {
+		return fmt.Errorf("missing %v", flags.L1EthRpcFlag.Name)
+	}
+	gameAddr, err := opservice.ParseAddress(ctx.String(GameAddressFlag.Name))
+	if err != nil {
+		return err
+	}
+	outputRoot, err := opservice.ParseAddress(ctx.String(OutputRootFlag.Name))
+	if err != nil {
+		return err
+	}
+	traceType := ctx.Uint64(TraceTypeFlag.Name)
+	l2BlockNum := ctx.Uint64(L2BlockNumFlag.Name)
+
+	l1Client, err := dial.DialEthClientWithTimeout(ctx.Context, dial.DefaultDialTimeout, logger, rpcUrl)
+	if err != nil {
+		return fmt.Errorf("failed to dial L1: %w", err)
+	}
+	defer l1Client.Close()
+
+	caller := batching.NewMultiCaller(l1Client.Client(), batching.DefaultBatchSize)
+	contract, err := contracts.NewFaultDisputeGameContract(gameAddr, caller)
+	if err != nil {
+		return fmt.Errorf("failed to create dispute game bindings: %w", err)
+	}
+
+	txMgrConfig := txmgr.ReadCLIConfig(ctx)
+	txMgr, err := txmgr.NewSimpleTxManager("challenger", logger, &metrics.NoopTxMetrics{}, txMgrConfig)
+	if err != nil {
+		return fmt.Errorf("failed to create the transaction manager: %w", err)
+	}
+
+	inner := contract.Inner()
+	methodCreateGame := "create"
+
+	call := inner.Call(methodCreateGame, uint32(traceType), outputRoot, l2BlockNum)
+	_, err = caller.SingleCall(context.Background(), batching.BlockLatest, call)
+	if err != nil {
+		return fmt.Errorf("failed to call create: %w", err)
+	}
+
+	call = inner.Call(methodCreateGame, uint32(traceType), outputRoot, l2BlockNum)
+	txCandidate, err := call.ToTxCandidate()
+	if err != nil {
+		return fmt.Errorf("failed to create tx candidate: %w", err)
+	}
+
+	rct, err := txMgr.Send(context.Background(), txCandidate)
+	if err != nil {
+		return fmt.Errorf("failed to send tx: %w", err)
+	}
+	fmt.Printf("Sent create transaction with status %v, tx_hash: %s\n", rct.Status, rct.TxHash.String())
+
+	call = inner.Call("getGameUUID", traceType, outputRoot, l2BlockNum)
+	uuid, err := caller.SingleCall(context.Background(), batching.BlockLatest, call)
+	if err != nil {
+		return fmt.Errorf("failed to call getGameUUID: %w", err)
+	}
+	gameUUID := uuid.GetHash(0)
+	fmt.Printf("Calculated Game UUID: %s\n", gameUUID.String())
```

Comment:
Why do we need the game UUID? I'd not an issue as such but seems like unnecessary code.